src/parser/xmlHandler.js

```diff
@@ -648,7 +648,14 @@ class XMLHandler {
         top.object = {};
       }
       if (top.object !== null) {
-        if (typeof top.object === 'object' && elementName in top.object) {
+        if (current.descriptor && current.descriptor.isMany) {
+          // Schema enforces array
+          if (!(elementName in top.object)) {
+            top.object[elementName] = [];
+          }
+          let val = top.object[elementName];
+          val.push(current.object);
+        } else if (typeof top.object === 'object' && elementName in top.object) {
           // The element exist already, let's create an array
```

Comment:
The block to check for isMany condition could be handled within the existing conditions isn't it ?

https://github.com/strongloop/strong-soap/blob/c40d08f2787aec03a7b6103f4d72e4bced2c410f/src/parser/xmlHandler.js#L651-L657

The reason I am asking is, as of now there are only two simple conditions in this block:
1. If the current element being parsed is already added to the wrapping element i.e., the top most in the stack.
        - this is to handle repetitive occurrences, sequences, etc of the current element
2. The current element should be added for the first time to its wrapping element
        - to link the current element to the wrapping element

You change seems to add an alternative path , just to check a value in the currently parsed element.

Reply:
> 2. The current element should be added for the first time to its wrapping element
>    - to link the current element to the wrapping element

The flaw in this point is that if the schema specifies that the client should expect possibly more than one (but also possibly one) element, the current element does not get wrapped in an array. I can surely make the logic change within existing paths if that is desired, but I hesitate as I am unsure if I want it to be so intertwined and hence risk the code being less easy to reason about.

See accompanying issue for further explanation as to why maxOccurs > 1 should be handled differently, even when the actual number of elements returned is 1.

Reply:
That's a fair comment - I've made the change as proposed, though I feel that it might make the resulting code slightly more difficult to reason about